src/ride/track.c

```diff
@@ -4880,15 +4881,587 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+    // jb      loc_6CDE4C
+    if (!map_can_construct_at(x_floored, y_floored, something, something_else, 0x0F)) {
+        // TODO: should this be inverted?
+        loc_6CDE4C();
+    }
+
+    // test    byte_F1AD60, 4
+    // jnz     loc_6CDE22
+    if (RCT2_GLOBAL(RCT2_ADDRESS_ELEMENT_LOCATION_COMPARED_TO_GROUND_AND_WATER, uint8) != 0x04) {
+        return loc_6CDE22();
+    }
+
+    // test    byte_F1AD60, 2
+    // jnz     loc_6CDE17
+    if (RCT2_GLOBAL(RCT2_ADDRESS_ELEMENT_LOCATION_COMPARED_TO_GROUND_AND_WATER, uint8) != 0x02) {
+        return loc_6CDE17();
+    }
+
+    // movzx   esi, byte ptr [esp+14h+var_10]
+    esi = RCT2_GLOBAL(edx, uint8);
+
+    // imul    esi, 260h
+    esi = esi * 0x260;
+
+    // push    eax
+    // push    edx
+    // push    edi
+    
+    // movzx   edi, rides[esi]
+    rct_ride *ride = GET_RIDE(esi);
+    
+    // movzx   eax, ds:word_97DD78[edi*4]
+    // imul    ds:dword_99DBC8
+    money32 price = RCT2_ADDRESS(0x0097DD78, money16)[ride->type * 4] * RCT2_GLOBAL(0x0099DBC8, money32);
+
+    // shrd    eax, edx, 10h
+    money32 temp_eax = (eax >> 0x10) + ((edx & 0xFFFF) << 0x10));
+
+    // sar     eax, 1
+    temp_eax = temp_eax / 2;
+
+    // imul    eax, 0Ah
+    temp_eax = temp_eax * 0x0A;
+
+    // mov     dword_F4413E, eax
+    RCT2_GLOBAL(0x00F4413E, money32) = temp_eax;
+
+    // pop     edi
+    // pop     edx
+    // pop     eax
+
+    // mov     ebx, [esp+14h+var_8]
+    ebx = ebx;
+
+    // test    bl, 1
+    if(((ebx & 0xFF) & 0x1) == 0) {
+        return loc_6CDDFD();
+    }
+    // jz      loc_6CDDFD
+
+    // or      byte ptr [esi+1], 10h
+
+    return loc_6CDAE8();
+}
+
+money32 loc_6CD99A(int eax, int ebx, int ecx, int edx, int esi, int edi, int ebp, int temp_eax, int temp_ecx) {
+    // mov     dx, di
+    uint16 temp_dx = (uint16) (edi & 0xFFFF);
+
+    // add     dx, 20h ; ' '
+    temp_dx += 0x20;
+
+    // shr     dx, 3
+    temp_dx = temp_dx >> 3;
+
+
+    // sub     dl, [ecx+2]
+    sint8 dl = (temp_dx & 0xFF) - RCT2_GLOBAL(temp_ecx + 2, sint8);
+
+    // jb      short loc_6CD9C5
+    if (dl < 0) {
+        return loc_6CD9C5(eax, ebx, ecx, edx, esi, edi, ebp);
+    }
+
+    // shr     dl, 1
+    dl = dl >> 1;
+
+    // cmp     dl, ds:byte_97D21A+9Eh
+    // jbe     short loc_6CD9C5
+    if (dl <= RCT2_GLOBAL(0x97D21A + 0x9E, uint8)) {
+        return loc_6CD9C5(eax, ebx, ecx, edx, esi, edi, ebp);
+    }
+
+    // pop     edx
+    // pop     ecx
+    // pop     eax
+
+    // mov     game_command_error_text, 1363
+    RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = STR_TOO_HIGH_FOR_SUPPORTS;
+
+    // jmp     loc_6CDE51
+    return loc_6CDE51();
+}
+
+money32 loc_6CD992(int eax, int ebx, int ecx, int edx, int esi, int edi, int ebp, int temp_eax, int temp_ecx) {
+    // add     ecx, 8
+    // test    byte ptr [ecx], 3Ch
+    // jnz     short loc_6CD992
+
+    do {
+        temp_ecx += 8;
+    } while ((RCT2_GLOBAL(temp_ecx, uint8) & 0x3C) != 0);
```

Comment:
This looks like a loop searching for the surface element. You can use map_get_surface_at. 

Note: The existing version assumes you will always find the surface element. The openRCT2 version does not assume that and can return null. Its up to you to decide how best to recover from this but I suggest just returning from the game_command with MONEY32_UNDEFINED